### PR TITLE
Updates bootstrap script to support differently named CSRs and external repos. Adds error handling to local-exec provisioners

### PR DESCRIPTION
**Pull request by timantrobus**
- Enables the bootstrap script to support cloning external private GitHub repos into differently named GCP Cloud Source Repositories (CSRs)
- Adds more robust logic to handle cloning of external repositories
- Enables conditional creation of SSH keys for GitHub machine user
- Refactors some Terraform `local-exec` provisioners to use bash script commands with additional error handling logic

### CODE REVIEW

**Review by memnac — APPROVED** [review]
Code reviewed and tested - working as expected!
Small enhancement to consider is to pass the REPO_URL value via a string variable or local and not directly. Generally a good practice for user-defined value that is subject to change.
Do you agree?

**variables.tf**
```
variable "policy_lib_repo_url" {
 type = string
 description = ...
 default = "https://github.com/forseti-security/policy-library.git"
}
variable "google_bootstrap_repo_url" {
 type = string
 description = ...
 default = "https://github.com/terraform-google-modules/terraform-google-bootstrap.git"
}
```
**main.tf**
```
REPO_URL = var.google_bootstrap_repo_url # line 286
REPO_URL = var.policy_lib_repo_url # line 328
```

**Review by memnac — APPROVED** [review]

File: bootstrap.sh [hunk 1]
```diff
@@ -286,19 +286,17 @@ function step5_generate_sshkeys_and_configure_git () {
...
     fi
 
-    # Generate keys if they don't exist
-    ssh-keygen -t rsa -b 4096 -N "" -q -C "${GITHUB_BOT_USER}" -f ~/.ssh/id_github > /dev/null # suppress output
-    ssh-keyscan -t rsa github.com | tee ~/.ssh/known_hosts > /dev/null # suppress output
-    check_exit_code $?
-
-    cat ssh_config_template > ~/.ssh/config
-
     printout "$(timestamp) [5-2]: Configure the cloudshell git session (email, name and credential)..."
```
Comment (memnac) [inline bootstrap.sh:300]: Small change here following testing - needs "\n"

`printout "\n$(timestamp) [5-2]: Configure the cloudshell git session (email, name and credential)..."`

Comment (timantrobus) [reply bootstrap.sh]: Thanks again for your review feedback @memnac. I have applied the suggested changes

**Comment by timantrobus** [comment]
> Code reviewed and tested - working as expected!
> Small enhancement to consider is to pass the REPO_URL value via a string variable or local and not directly. Generally a good practice for user-defined value that is subject to change.
> Do you agree?
> 
> **variables.tf**
> 
> ```
> variable "policy_lib_repo_url" {
>  type = string
>  description = ...
>  default = "https://github.com/forseti-security/policy-library.git"
> }
> variable "google_bootstrap_repo_url" {
>  type = string
>  description = ...
>  default = "https://github.com/terraform-google-modules/terraform-google-bootstrap.git"
> }
> ```
> 
> **main.tf**
> 
> ```
> REPO_URL = var.google_bootstrap_repo_url # line 286
> REPO_URL = var.policy_lib_repo_url # line 328
> ```

Added these variable substitutions as suggested.

**Pull request merged**